### PR TITLE
Improve documentation of GFRE LFRE and OrthogonalRegression notebooks

### PR DESCRIPTION
**Pull request by agoscinski**
As the title says, the documentation of the notebooks has been reworked, the plots have been made a bit more readable.

### CODE REVIEW

File: examples/PlotLFRE.ipynb [hunk 1]
```diff
@@ -222,7 +141,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.6.11"
+   "version": "3.7.6"
```
Comment (agoscinski) [inline examples/PlotLFRE.ipynb:144]: I guess this is a change of the newer notebook version I am using. I am not sure if I should just remove it, or change it to something consistent. Do you have any thoughts?

Comment (rosecers) [reply examples/PlotLFRE.ipynb:144]: You should remove these sorts of edits from the notebooks before merging. Do you generally make sure to strip the NBs?

File: examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb [hunk 2]
```diff
@@ -4,7 +4,28 @@
...
+    "\n",
+    "with the constraint that $\\Omega$ is an orthogonal projection/matrix  $\\Omega^T\\Omega=I$. In this formulation it can be only solved if $y$ and $X$ have the same dimension, i.e. they agree in the number of features. If `use_orthogonal_projector` is set to False, the feature dimension of $X$ or $y$ are padded with 0 values such that they agree in the number of dimension. To be able to compute an orthogonal projector without any padding of the matrices the option `use_orthogonal_projector` is offered with the caveat that it does not always well behave analytically. For this approach we take the linear regression solution $W$\n",
+    "\n",
+    "$$ \\min_W ||y - XW\\||_F,$$\n",
+    "\n",
+    "applyt SVD on it $W = USV^T$ and solve the Procrustes problem for\n",
+    "\n",
+    "$$\\min_\\Omega' ||yV - XU\\Omega'\\||_F\\quad \\Omega'^T\\Omega'=I$$\n",
+    "\n",
+    "The final orthogonal projector is then $\\Omega =  U\\Omega' V^T$. This solution can however be problematic, since changing the number of features can result in different predictions. This notebook shows an example of this problematic behaviour. "
```
Comment (rosecers) [inline examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb:28]: This set of sentences doesn't make much sense. From a simple copy-editing standpoint, it should read:

> In this example, we explain how when using `skcosmo.linear_model.OrthogonalRegression` the option `use_orthogonal_projector` can result in non-analytic behavior. 
> In `skcosmo.linear_model.OrthogonalRegression`, we solve the linear regression problem assuming an orthogonal weighting matrix $\\Omega$ to project from the feature space $X$ to the target space $y$.
> $$\\min_\\Omega ||y - X\\Omega\\||_F$$
> This assumes that $X$ and $y$ contain the same number of features. 
> If `use_orthogonal_projector=False`, the smaller of $X$ and $y$ is padded with null features, i.e. columns of zeros. 
> However, when `use_orthogonal_projector=True`, we begin with the weights $W$ determined by the linear regression problem
> $$ \\min_W ||y - XW\\||_F,$$
> and solve the Procrusted problem for 
> $$\\min_\\Omega' ||yV - XU\\Omega'\\||_F\\quad \\Omega'^T\\Omega'=I$$
> where the SVD of $W = USV^T$
> The final orthogonal projector is then $\\Omega =  U\\Omega' V^T$. 
> In this notebook, we demonstrate a problem that may arise with this solution, as changing the number of features can result in different predictions.

File: examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb [hunk 3]
```diff
@@ -17,8 +38,11 @@
...
     "from skcosmo.linear_model import OrthogonalRegression\n",
     "\n",
+    "# These are coordinates of a 3-dimensional cube. We treat the points of the cube as samples\n",
+    "# and the 3 dimensions as features x y z\n",
     "cube = np.array(\n",
     "    [\n",
+    "        #x  y  z\n",
     "        [0, 0, 0],\n",
     "        [1, 0, 0],\n",
     "        [0, 1, 0],\n",
```
Comment (rosecers) [inline examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb:48]: Pet peeve -- you should swap the order of the 4th and 5th vertex

Comment (rosecers) [reply examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb:48]: Same with the other cube below.

File: examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb [hunk 4]
```diff
@@ -30,12 +54,11 @@
...
     "\n",
-    "# Can be seen as a cube with zero z-scaling without the z-dimension.\n",
-    "# This is important because we compare it with a cube with zero\n",
-    "# z-scaling keeping the z-dimension\n",
-    "xy_plane_projected_cube = cube[:,[0,1]]\n",
     "\n",
+    "# the x y coordinates of the cube\n",
+    "xy_plane_projected_cube = cube[:, [0, 1]]\n",
     "\n",
+    "# cube with \n",
```
Comment (rosecers) [inline examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb:61]: This is called a square prism, once z!=1, it is no longer a cube

File: examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb [hunk 5]
```diff
@@ -50,15 +73,15 @@
...
+    "# In terms of regressable information `xy_plane_projected_cube` can be seen as z_scaled_cube with z_scaling = 0.\n",
+    "# Adding features with zero values should in general not change the prediction quality"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We compute the orthogonal regression error for the orthogonal regression of a cube fitting on a cube with z-scaling in the range [0,1]. For the case of a zero z-scaling the error is computed one time with a third dimension and one time without. The regression is done with `skcosmo.linear_model.OrthogonalRegression` first setting the `use_orthogonal_projector` to True."
+    "We compute now the orthogonal regression error fitting on a cube with z-scaling in the range of [0,1] to predict the cube. For the case of a zero z-scaling the error is computed one time with a third dimension and one time without it (using `xy_plane_projected_cube`). The regression is done with `skcosmo.linear_model.OrthogonalRegression` by first setting the `use_orthogonal_projector` to True."
```
Comment (rosecers) [inline examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb:84]: More editing for clarity.
```suggestion
    "We now compute the orthogonal regression error fitting on the square prism to predict the cube. In the case of a zero z-scaling, the error is computed once with a third dimension and once without it (using `xy_plane_projected_cube`). The regression is done with `skcosmo.linear_model.OrthogonalRegression`  `use_orthogonal_projector` set to True."
```

File: examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb [hunk 6]
```diff
@@ -50,15 +73,15 @@
     "        ]\n",
     "    )\n",
     "\n",
-    "\n",
-    "z_scalings = np.linspace(0, 1, 10)"
+    "# In terms of regressable information `xy_plane_projected_cube` can be seen as z_scaled_cube with z_scaling = 0.\n",
```
Comment (rosecers) [inline examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb:76]: This sentence is non-sensical and either should be removed or rephrased.

**Review by rosecers — CHANGES_REQUESTED** [review]
@agoscinski, these still need some work to go from syntax examples to demonstrative examples. In the OrthogonalRegression example, given that you are working with a relatively easy-to visualize solid, I'd like to see more plotting of the results, along the line of

```
fig, (ax_xy, ax_xz, ax_yz) = plt.subplots(1, 3, figsize=(12, 4))
ax_xy.scatter(orth_reg_pred_cube[:, 0], orth_reg_pred_cube[:, 1], c='r')
ax_xz.scatter(orth_reg_pred_cube[:, 0], orth_reg_pred_cube[:, 2], c='r')
ax_yz.scatter(orth_reg_pred_cube[:, 1], orth_reg_pred_cube[:, 2], c='r')
ax_xy.scatter(cube[:, 0], cube[:, 1], c='b')
ax_xz.scatter(cube[:, 0], cube[:, 2], c='b')
ax_yz.scatter(cube[:, 1], cube[:, 2], c='b')
```

which shows how the cube is rotated in 3D space when regression.

In GFRE and LFRE, you need to break up your blocks of code with comments and you do not need the status print statements. I'd much rather the second block in GFRE look like this:

```
gfre_matrix = np.zeros((2, 2))

# self-GFRE of the SOAP features
gfre_matrix[0, 0] = global_reconstruction_error(soap_features, soap_features)

# GFRE of the SOAP features on the bispectrum features
gfre_matrix[0, 1] = global_reconstruction_error(soap_features, bispectrum_features)

# GFRE of the bispectrum features on the SOAP features
gfre_matrix[1, 0] = global_reconstruction_error(bispectrum_features, soap_features)

# self-GFRE of the bispectrum features
gfre_matrix[1, 1] = global_reconstruction_error(
    bispectrum_features, bispectrum_features
)
```

This may seem redundant but really helps a reader follow the example. You may also want to consistently call the power spectrum the *power spectrum*, as opposed to the SOAP_features, as both are listed as SOAP-type features somewhere. Remember that a user may not be as intimately aware of the SOAP spectrum as you are.

**Comment by agoscinski** [comment]
okay open for another review

File: examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb (hunk 6 above)
Comment (rosecers) [reply examples/OrthogonalRegressionNonAnalytic_no-doc.ipynb]: Much better

**Comment by rosecers** [comment]
@agoscinski I made some formatting changes to clean stuff up but am not able to push directly to your branch. Please take a look at https://github.com/cosmo-epfl/scikit-cosmo/tree/pr/95 and merge them onto your branch if you see fit. After that, I think we're good to merge!

By the way, I very much like the new OrthogonalRegression example, so I included it in the RTD in this branch.

**Review by rosecers — APPROVED** [review]

**Comment by rosecers** [comment]
Just make sure to squash before merge!

**Pull request merged**